From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 25 Apr 2023 17:43:16 +0200
Subject: [PATCH] make sure cargo is called with --frozen and the correct
 --target option

PTXdist defines CARGO_BUILD_TARGET when this is needed.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 cargo_wrapper.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cargo_wrapper.py b/cargo_wrapper.py
index e04ad39f2dfc..c9e6885d4133 100644
--- a/cargo_wrapper.py
+++ b/cargo_wrapper.py
@@ -100,6 +100,9 @@ if __name__ == "__main__":
         print("Unknown command:", opts.command, file=logfile)
         sys.exit(1)
 
+    if 'CARGO_BUILD_TARGET' in env:
+        cargo_cmd += ['--target', env['CARGO_BUILD_TARGET'], '--frozen']
+
     if features:
         cargo_cmd += ['--features', ','.join(features)]
     cargo_cmd += ['--target-dir', cargo_target_dir]
